addons/mail/static/src/models/thread/thread.js

```diff
@@ -1239,6 +1293,20 @@ function factory(dependencies) {
             return [['replace', this.messages.filter(message => message.isNeedaction)]];
         }
 
+        /**
+         * @returns {integer|undefined}
+         * @private
+         */
+        _computeMessageBeforeNewMessageSeparatorId() {
+            if (this.model !== 'mail.channel') {
+                return clear();
+            }
+            if (this.localMessageUnreadCounter === 0) {
+                return clear();
+            }
+            return this.lastSeenByCurrentPartnerMessageId;
+        }
```

Comment:
Why not a relation? We should avoid attribute derived from another record attribute. Keep dealing with just records when you can, and avoid going too deep with fields.

Reply:
As per field doc, this is using `undefined` for "no separator" and `0` for separator at the start of the list. This distinction is not possible with relation.

Reply:
IMO this distinction of 2 sentinel values should not exist.
No separator should happen when there are no new message, and separator at the start of list should happen when message just before separator is older than the oldest message displayed.